src/Components/Web/src/Virtualization/ItemsProviderDelegate.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Virtualization
+{
+    /// <summary>
+    /// A function that provides items to a virtualized source.
+    /// </summary>
+    /// <typeparam name="TItem">The type of the context for each item in the list.</typeparam>
+    /// <param name="request">The <see cref="ItemsProviderRequest"/> defining the request details.</param>
+    /// <returns>A <see cref="Task"/> whose result is a <see cref="ItemsProviderResult{TItem}"/> upon successful completion.</returns>
+    public delegate Task<ItemsProviderResult<TItem>> ItemsProviderDelegate<TItem>(ItemsProviderRequest request);
```

Comment:
Sorry I should have thought of this before, but now I see the implementation it's a bit clearer! What do you think about declaring the result type as `ValueTask<ItemsProviderResult<TItem>>`? This would eliminate an unnecessary allocation in the case where the developer can return the completed result data synchronously.